### DISCUSSION TITLE
Keycloak IDP Initiated flow to launch java-saml-tookit-jspsample

### DISCUSSION THREAD

**Discussion by prashantmdesai** [keycloak/keycloak#17004] [Q&A]
I have configured the java-saml-tookit-jspsample application from below github repo to my tomcat server.
https://github.com/SAML-Toolkits/java-saml

I am trying to configure KeyCloak for IDP initiated flow. (the SP initiated flow is already working with successful login page visibility and http post after that to the ACS).

In case of IDP initiated flow, I am getting "Page not found" error when the user clicks on the application URL in the KeyCloak account/#/applications page: http://localhost:8082/realms/IDP_REALM/account/#/applications

That means, the IDP is not doing the HTTP post to the target URL. I already verified using saml-tracer that the request going to the target URL is not containing the SAML assertion.

What am I missing? Please help.
This is my ACS url: http://localhost:8080/java-saml-tookit-jspsample/acs.jsp

I also tried to access my target URL directly by browsing this URL: http://localhost:8082/realms/IDP_REALM/protocol/saml/clients/http://localhost:8080/java-saml-tookit-jspsample/acs.jsp
I get "Page not found" error for this as well.

Please let me know if anybody faced this.

Is my target IDP initiated SSO URL wrong? http://localhost:8082/realms/IDP_REALM/protocol/saml/clients/http://localhost:8080/java-saml-tookit-jspsample/acs.jsp

Attached is my onelogin.saml.properties file.
[onelogin.saml.properties.txt](https://github.com/keycloak/keycloak/files/10712893/onelogin.saml.properties.txt)

Below is my client config in kc

![image](https://user-images.githubusercontent.com/11394924/218246947-f092b522-e0c8-403e-ba5d-57f67c97f75c.png)

![image](https://user-images.githubusercontent.com/11394924/218246960-a35f2e32-7969-4d25-88fd-2b5744bff784.png)

![image](https://user-images.githubusercontent.com/11394924/218246971-30e79567-2e3f-4364-b41a-8cc8d88c0c76.png)

**Comment by UmarAalam** [comment]
possible reasons:
1. check locally if jsp file redirection is accessible or not. (acs.jsp)
2. **idp.x509cert** mentioned in onelogin.saml.properties file is in multiline---> change it to single line.
